plainly-aescripts/src/validateProject.ts

```diff
@@ -0,0 +1,46 @@
+interface ProjectValidation {
+  textLayers?: {
+    allCaps: {
+      layerId: string;
+      layerName: string;
+    }[];
+  };
+}
+
+function checkTextLayers(): ProjectValidation['textLayers'] | undefined {
```

Comment:
`ProjectValidation['textLayers']` this is a bit weird, you can make something like this:

```ts
enum ProjectIssueType {
    AllCaps,
    SomethingElse
}

type ProjectIssue<T extends ProjectIssueType> = {
    type: T;
}

type ProjectLayerIssue<T extends ProjectIssueType> = ProjectIssue<T> & {
    layerPath: {id: string, name: string}[]; // so you can find it in the project
    layerName: string;
}


type TextAllCapsEnabledIssue = ProjectLayerIssue<ProjectIssueType.AllCaps>;

type AnyProjectIssue = TextAllCapsEnabledIssue;


// validate project
const projectIssues: AnyProjectIssue[] = checkProject(app.project);
```